### PR TITLE
[WIP] adding azure firewall modules

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
This is to add Azure Firewall modules.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
azure_rm_azurefirewall
azure_rm_azurefirewall_info

### CODE REVIEW

**Comment by yuwzho** [comment]
Fixes #54440

File: lib/ansible/module_utils/azure_rm_common_ext.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                result['compare'] = 'changed [' + path + '] length is different or null'
+                return False
+            if isinstance(old[0], dict):
+                key = None
+                if 'id' in old[0] and 'id' in new[0]:
+                    key = 'id'
+                elif 'name' in old[0] and 'name' in new[0]:
+                    key = 'name'
+                else:
+                    key = list(old[0].keys())[0]
```
Comment (abadger) [inline lib/ansible/module_utils/azure_rm_common_ext.py]: This is more efficient:
```suggestion
                    key = next(iter(old[0])
```

Comment (zikalino) [reply lib/ansible/module_utils/azure_rm_common_ext.py]: thanks!

**Comment by zikalino** [comment]
closing old one as too many commits

**Pull request closed without merging**